tests/unit/component/distribution_policy_executor.cpp

```diff
@@ -0,0 +1,56 @@
+//  Copyright (c) 2015 Hartmut Kaiser
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/hpx_main.hpp>
+#include <hpx/include/components.hpp>
+#include <hpx/include/actions.hpp>
+#include <hpx/include/async.hpp>
+#include <hpx/include/util.hpp>
+#include <hpx/include/parallel_executors.hpp>
+#include <hpx/util/lightweight_test.hpp>
+
+///////////////////////////////////////////////////////////////////////////////
+hpx::id_type call() { return hpx::find_here(); }
+
+HPX_PLAIN_ACTION(call, call_action);
+
+struct call_pfo
+{
+    hpx::id_type operator()() const
+    {
+        return hpx::find_here();
+    }
+};
+
+///////////////////////////////////////////////////////////////////////////////
+void test_distribution_policy_executor()
+{
+    using namespace hpx::parallel;
+
+    for (hpx::id_type const& loc: hpx::find_all_localities())
+    {
+        auto exec = make_distribution_policy_executor(hpx::colocated(loc));
+        typedef executor_traits<decltype(exec)> executor_traits;
+
+        HPX_TEST_EQ(executor_traits::async_execute(exec, call_pfo()).get(), loc);
```

Comment:
See: https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/runtime/serialization/access.hpp#L66. We are able to serialize empty types without explicit serialization support.